compiler/rustc_ast_passes/src/ast_validation.rs

```diff
@@ -365,46 +366,94 @@ impl<'a> AstValidator<'a> {
         }
     }
 
-    /// An `extern "custom"` function must be unsafe, and must not have any parameters or return
-    /// type.
-    fn check_custom_abi(&self, ctxt: FnCtxt, ident: &Ident, sig: &FnSig) {
+    /// Check that this function does not violate the constraints of its ABI.
+    fn check_abi(&self, abi: ExternAbi, ctxt: FnCtxt, ident: &Ident, sig: &FnSig) {
```

Comment:
I find these `check_this` names to be slightly nondescript.
1. sometimes they warn. sometimes they error.
2. we specifically check only the signature.
```suggestion
    fn reject_invalid_abi_sig(&self, abi: ExternAbi, ctxt: FnCtxt, ident: &Ident, sig: &FnSig) {
```

Reply:
I did some renaming. Most functions in that module do start with `check_` so that is what I followed.